### ISSUE TITLE
Add a way to trace something once

### ISSUE THREAD

**Issue by LucPrestin** [LucPrestin/Hidden-Modularity#12]
As a user of the TraceDebugger, I'd like to be able to trace code only once so that I don't have a big number of trace debuggers open for things that happen often in a program execution.

---

**Describe the solution you'd like**

Some things, especially in graphics output get called often. One example is `Morph>>drawOn:`. In that special case a multitude of TraceDebuggers opens, whenever something draws itself over the morph. Here it would be nice to only do the trace on the first call and just execute the code the other times, without tracing it.

**Hints & Thoughts**

- event filters could help when tracing events
  - every event goes through the `handMorph`. We can use event filters here with special conditions like *it is a mouse event and a mouse click and this global variable is false*
  - the API is documented in `Morph>>eventFilterDocumentation`
  - an exemplary implementation can be found in `LegacyShortcutsFilter`

---

## Conditions of satisfaction

- [ ] Should trace code only once
- [ ] Should have a mechanism to reset the check whether the code was already called so that the trace can be re-done if wanted

**Comment by LucPrestin** [comment]
The TraceDebugger allows to trace the execution of a block enclosure. There are enough things we can do with that, so this Issue is on ice for now.

**Comment by LinqLover** [comment]
It should be very easy to build something like `traceAndDebugOnce` similar to `Object>>#haltOnce`:

```smalltalk
haltOnce
	"Halt unless we have already done it once."
	self haltOnceEnabled
		ifTrue: [self clearHaltOnce.
			^ self halt]
```
